extensions/dropstab_raycast/src/search-crypto.tsx

```diff
@@ -0,0 +1,140 @@
+      {!state.query && favorites.length > 0 && (
+        <List.Section title="Favorites:">
+          {favorites.map((favorite) => (
+            <SearchListItem
+              key={favorite.id}
+              searchResult={favorite}
+              favorites={favorites}
+              setFavorites={setFavorites}
+              addToFavorites={(item) => addToFavorites(item, favorites, setFavorites, CRYPTO_FAVORITES_KEY)}
+              removeFromFavoritesForCoins={(item) =>
+                removeFromFavoritesForCoins(item, favorites, setFavorites, CRYPTO_FAVORITES_KEY)
+              }
+              isFavoriteForCoins={(item) => isFavoriteForCoins(item, favorites)}
+            />
+          ))}
+        </List.Section>
+      )}
+      <List.Section title={state.query ? "Results:" : "Trending:"}>
+        {state.results &&
+          state.results.map((searchResult) => (
+            <SearchListItem
+              key={searchResult.id}
+              searchResult={searchResult}
+              favorites={favorites}
+              setFavorites={setFavorites}
+              addToFavorites={(item) => addToFavorites(item, favorites, setFavorites, CRYPTO_FAVORITES_KEY)}
+              removeFromFavoritesForCoins={(item) =>
+                removeFromFavoritesForCoins(item, favorites, setFavorites, CRYPTO_FAVORITES_KEY)
+              }
+              isFavoriteForCoins={(item) => isFavoriteForCoins(item, favorites)}
+            />
+          ))}
+      </List.Section>
+    </List>
+  );
+}
+
+function SearchListItem({
+  searchResult,
+  favorites,
+  setFavorites,
+  addToFavorites,
+  removeFromFavoritesForCoins,
+  isFavoriteForCoins,
+}: {
+  searchResult: SearchResult;
+  favorites: SearchResult[];
+  setFavorites: (favorites: SearchResult[]) => void;
+  addToFavorites: (
+    item: SearchResult,
+    favorites: SearchResult[],
+    setFavorites: (favorites: SearchResult[]) => void,
+    key: string,
+  ) => void;
+  removeFromFavoritesForCoins: (
+    item: SearchResult,
+    favorites: SearchResult[],
+    setFavorites: (favorites: SearchResult[]) => void,
+    key: string,
+  ) => void;
+  isFavoriteForCoins: (item: SearchResult) => boolean;
+}) {
+  if (!searchResult || !searchResult.id) {
+    return null;
+  }
+
+  return (
+    <List.Item
+      key={searchResult.id}
+      icon={{ source: searchResult.icon || "", mask: Image.Mask.RoundedRectangle }}
+      title={`${searchResult.symbol}`} // Изменено для отображения rank перед символом
+      subtitle={searchResult.name}
+      accessories={[{ text: `${searchResult.rank} | ${searchResult.price} | ${searchResult.marketCap}` }]}
+      actions={
+        <ActionPanel>
+          <ActionPanel.Section>
+            <Action.OpenInBrowser title="Open in Dropstab" url={searchResult.url} icon="dropstab-logo.png" />
+            {isFavoriteForCoins(searchResult) ? (
+              <ActionPanel.Item
```

Comment:
```suggestion
              <Action
```
`<ActionPanel.Item` is deprecated